hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/S3APerformanceFlags.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a.impl;
+
+import java.util.Locale;
+
+import org.apache.hadoop.fs.StreamCapabilities;
+
+import static org.apache.hadoop.fs.s3a.Constants.FS_S3A_CREATE_PERFORMANCE_ENABLED;
+import static org.apache.hadoop.fs.s3a.Constants.FS_S3A_PERFORMANCE_FLAGS;
+
+/**
+ * Performance flags.
+ * These are stored as a map of options.
+ */
+public final class S3APerformanceFlags implements StreamCapabilities {
+
+  /**
+   * Flag for create performance: {@value}.
+   */
+  public static final String CREATE = "create";
+
+  /**
+   * Flag for delete performance: {@value}.
+   */
+  public static final String DELETE = "delete";
+
+  /**
+   * Flag for mkdir performance: {@value}.
+   */
+  public static final String MKDIR = "mkdir";
+
+  /**
+   * Enable all performance flags: {@value}.
+   */
+  public static final String ALL = "*";
+
+  /**
+   * Higher performance create operations.
+   */
+  private boolean create;
+
+  /**
+   * Delete operation to skip parent probe.
+   */
+  private boolean delete;
+
+  /**
+   * Mkdir to skip checking for type of parent paths.
+   */
+  private boolean mkdir;
+
+  public S3APerformanceFlags() {
+  }
+
+  public boolean isCreate() {
+    return create;
+  }
+
+  public boolean isDelete() {
+    return delete;
+  }
```

Comment:
This one also we want to tackle as separate task (similar to HADOOP-19072), correct?

Reply:
yes. harshit did an experiment where he turned off all attempts at creating parent dirs after delete. fairly brittle, i think